### DISCUSSION TITLE
NO DATA with bluetooth ODBlink LX

### DISCUSSION THREAD

**Discussion by mikelesagitter** [cedricp/ddt4all#1199] [Q&A]
Hi there, I saw that I'm not the only one having the NO DATA issue. Other people solved by redoing the cable modification but I use a bluethooth dongle... I don'y use a cable, so what's the issue in my case?

Here you have the logs, hope it will help

I'm triyng to activate Android Auto in a 2013 zoe. When I go to the page ECU configuration, I see this:

![Capture](https://github.com/user-attachments/assets/6a73988f-63f3-4e8e-b521-be19438f204c)

I first did a "read all", then a write ECU config...

`UCE utilise diagnostic session 10C0
Envoie requête :StartAfterSalesSession
Retard 50 ms
Changement de mode session 10 C0
Envoie requête :DataRead.ParkAssist
Retard 0 ms
Envoie requête ELM :22 6C 0D
Réponse ELM :  
Envoie requête :DataRead.Vehicle_type_configuration
Retard 0 ms
Envoie requête ELM :22 6C 61
Réponse ELM :  
Envoie requête :DataRead.ADAS
Retard 0 ms
Envoie requête ELM :22 6C 17
Réponse ELM :  
Envoie requête :DataRead.UPA_MuteTimeout
Retard 0 ms
Envoie requête ELM :22 6C 19
Réponse ELM :  
Envoie requête :DataRead.HVAC
Retard 0 ms
Envoie requête ELM :22 6C 10
Réponse ELM :  
Envoie requête :DataRead.Emergency_call_number
Retard 0 ms
Envoie requête ELM :22 6C 26
Réponse ELM :  
Envoie requête :DataRead.TCU
Retard 0 ms
Envoie requête ELM :22 6C 18
Réponse ELM :  
Envoie requête :DataRead.TelematicServices
Retard 0 ms
Envoie requête ELM :22 6C 4C
Réponse ELM :  
Envoie requête :DataRead.Vehicle charging receptacle type
Retard 0 ms
Envoie requête ELM :22 6C 5E
Réponse ELM :  
Envoie requête :DataRead.Vehicle Charging Cable type1
Retard 0 ms
Envoie requête ELM :22 6C 5F
Réponse ELM :  
Envoie requête :DataRead.Vehicle Charging Cable type2
Retard 0 ms
Envoie requête ELM :22 6C 60
Réponse ELM :  
Envoie requête :DataRead.Engine_configuration
Retard 0 ms
Envoie requête ELM :22 6C 13
Réponse ELM :  
Envoie requête :DataRead.VehicleSetting
Retard 0 ms
Envoie requête ELM :22 6C 0F
Réponse ELM :  
Envoie requête :DataRead.ECU
Retard 0 ms
Envoie requête ELM :22 6C 1C
Réponse ELM :  
Envoie requête :StartDiagnosticSession
Retard 250 ms
Changement de mode session 10 81
Envoie requête :StartAfterSalesSession
Retard 50 ms
Changement de mode session 10 C0
Envoie requête :DataRead.ParkAssist
Retard 0 ms
Envoie requête ELM :22 6C 0D
Réponse ELM :  
Envoie requête :DataRead.Vehicle_type_configuration
Retard 0 ms
Envoie requête ELM :22 6C 61
Réponse ELM :  
Envoie requête :DataRead.ADAS
Retard 0 ms
Envoie requête ELM :22 6C 17
Réponse ELM :  
Envoie requête :DataRead.UPA_MuteTimeout
Retard 0 ms
Envoie requête ELM :22 6C 19
Réponse ELM :  
Envoie requête :DataRead.HVAC
Retard 0 ms
Envoie requête ELM :22 6C 10
Réponse ELM :  
Envoie requête :DataRead.Emergency_call_number
Retard 0 ms
Envoie requête ELM :22 6C 26
Réponse ELM :  
Envoie requête :DataRead.TCU
Retard 0 ms
Envoie requête ELM :22 6C 18
Réponse ELM :  
Envoie requête :DataRead.TelematicServices
Retard 0 ms
Envoie requête ELM :22 6C 4C
Réponse ELM :  
Envoie requête :DataRead.Vehicle charging receptacle type
Retard 0 ms
Envoie requête ELM :22 6C 5E
Réponse ELM :  
Envoie requête :DataRead.Vehicle Charging Cable type1
Retard 0 ms
Envoie requête ELM :22 6C 5F
Réponse ELM :  
Envoie requête :DataRead.Vehicle Charging Cable type2
Retard 0 ms
Envoie requête ELM :22 6C 60
Réponse ELM :  
Envoie requête :DataRead.Engine_configuration
Retard 0 ms
Envoie requête ELM :22 6C 13
Réponse ELM :  
Envoie requête :DataRead.VehicleSetting
Retard 0 ms
Envoie requête ELM :22 6C 0F
Réponse ELM :  
Envoie requête :DataRead.ECU
Retard 0 ms
Envoie requête ELM :22 6C 1C
Réponse ELM :  
Envoie requête :StartDiagnosticSession
Retard 250 ms
Changement de mode session 10 81
Envoie requête :StartAfterSalesSession
Retard 50 ms
Changement de mode session 10 C0
Envoie requête :DataWrite.ECU
Retard 250 ms
Envoie requête ELM :2E 6C 1C E4
Réponse ELM :  
Envoie requête :StartDiagnosticSession
Retard 250 ms
Changement de mode session 10 81`

**Comment by Furtif** [comment]
i think LX not ok https://github.com/cedricp/ddt4all?tab=readme-ov-file#supported-diagnostic-adapters-so-far

i have tested SX and EX and fine but not added (EX in readme so far)

**Comment by AndyIreland** [comment]
> Hi there, I saw that I'm not the only one having the NO DATA issue. Other people solved by redoing the cable modification but I use a bluethooth dongle... I don'y use a cable, so what's the issue in my case?
> 
> Here you have the logs, hope it will help
> 
> I'm triyng to activate Android Auto in a 2013 zoe. When I go to the page ECU configuration, I see this:
> 
> ![Capture](https://private-user-images.githubusercontent.com/6686471/358674203-6a73988f-63f3-4e8e-b521-be19438f204c.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM4MzcxMTQsIm5iZiI6MTcyMzgzNjgxNCwicGF0aCI6Ii82Njg2NDcxLzM1ODY3NDIwMy02YTczOTg4Zi02M2YzLTRlOGUtYjUyMS1iZTE5NDM4ZjIwNGMuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTZUMTkzMzM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDYxMGY3MWE3YzY5MDNmMDA3OWNiZmFhZDM2MTRkYzMyMTRlOWI1NDFiNmRhMjAwNTA2ZDViYzBlYzIzOWQ2NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.rwEyWm9W_mXTUEidLflf-a0GLxUzvDsB-lzI5GNNDpk)
> 
> I first did a "read all", then a write ECU config...


Sooo. it sounds like you haven't wired up a cable or modified your elm327 by re-soldering it so you can read your rlink through the obd2 socket ... have you? @mikelesagitter

**Reply by mikelesagitter** [reply]
No, I haven't... so even wireless obd needs to be soldered? I just need to do a bridge between 2 pins or have to buy an extention cord to switch cables between the car port and the obd?

2 days past I bought 2 more OBD, a wifi elm and an USB 327 with pic18 chip. Guess I should try to solder one of them... I wanted to avoid this so much :D

**Reply by AndyIreland** [reply]
yes I am afraid whether wireless or wired (USB) you still have to make the modification on the elm327 itself (by solder) or get a y-cable and make the adjustments so you can read multimedia (rlink) through the obd2 socket  -  every other ecu you can read/write without making this type of adjustment. 

https://www.gps-rlink.com/r-link-1/activate-android-auto-waze-on-r-link-1/

![image](https://github.com/user-attachments/assets/051b4522-f05d-4f8c-9604-34265f83be7d)

**Reply by mikelesagitter** [reply]
![20240819_001504](https://github.com/user-attachments/assets/6ea5a02b-d962-44cb-87c3-74ef44b5e4e0)

I'm testing my skills ont the obdlink LX to see if I'm doing right. Like this? And the other side of the 13 goes on the 3

**Reply by AndyIreland** [reply]
with mine I went the wiring modifications on a y-splitter cable , I know it looks messy but it worked:
![image](https://github.com/user-attachments/assets/413201ba-f3f1-4a11-b017-afec5b75cb30)

**Comment by AndyIreland** [comment]
@mikelesagitter  -  DONT WHATEVER YOU DO PRESS THE '**SOFT RESET**' BUTTON IN DDT4ALL when enabling Android Auto !!

**Reply by mikelesagitter** [reply]
I found a tutorial that tells to do a soft reset after enabling android auto 🤣 ok so what I did with soldering looks good?

**Reply by mikelesagitter** [reply]
Just out of curiosity, what happens if I soft reset after installing android auto?

**Reply by AndyIreland** [reply]
noooooooo, just dont do it! LOL - if you press reset soft button you loose all settings on your r-link radio and get a black screen with a triangle on it and nothing works (semi bricked!) - you can in most cases get it working again by filling out all the parameters and sending a .cmd script to it .... but just dont even get to that stage . 

You can if you want to press the 'Reset Hard'  button (after all the lights apart from the ELM327 dongle have stopped flashing, and just the power light is steady) to initiate changes or press the home button next to the touch screen 5 times to re-boot your r-link and it should start up again and when you go into the menu an 'ANDROID AUTO' or Mirror link button should show up on screen (If you have enabled AA properly)

**Reply by mikelesagitter** [reply]
I tested the OBD link with wire bridges and it does not work :D I still dont get if can only do those bridges or do I have to buy a cord and invert the wires on it. Can you tell me if the 2 methods are the same or I need to do the cord one?

**Reply by AndyIreland** [reply]
the ELM327 you have (OBD2 dongle) has it got a PIC18F25K80 chip in it , or is it a cheap chinese clone ELM?

when you say 'it did not work' - when you made wiring adjustments where there still red boxes with 'NO DATA' in them or did it read the parameters right?

If you got as far as changing the parameters in DDT4all and enabled AA , did you 'hard reset' the rlink afterwards to reboot?

**Reply by mikelesagitter** [reply]
No, I just made a bridge between the pins like I showed but it was on the Obdlink SX, witch Cedric has told not compatible. But I tried anyway on the car and ddt4all didn't even connect to it (all the lights where blinking as a Christmas tree...) so now I don't know if the soldering I did (bridge between the pins) is the good method or if I would buy a cord and invert the wires (I'm not so good at electronic stuff). Either way I bought a usb elm with the right chip and I'll modify it once you tell minif the bridge is the right method :)

**Reply by AndyIreland** [reply]
hard to tell from your photo. I just wired up a y-cable for myself and used a ELM with correct chip in it and it worked. When I want to change parameters on the r-link I just pop the ELM into the Y-Cable , when I just want to adjust all other ECU's then I just plug the ELM straight into the OBD2 socket - Simpelz! :)

**Reply by mikelesagitter** [reply]
Ok I get it... The thing is I don't know if electronically speaking the bridge and the wire switching is the same thing :) but I'll see If a find a cable

**Reply by mikelesagitter** [reply]
![20240819_154521](https://github.com/user-attachments/assets/81b912c0-c563-49f6-be22-57b536641476)
 Here is my usb odb module. I wonder what's the switch for! It doesn't turn off or on...

**Reply by AndyIreland** [reply]
that looks like its a FORSCAN ELM327 for Ford vehicles where you switch for CAN-L and CAN-H , its wired up differently as to what you would wire up a ELM327 for enabling AA on the Renault r-link.

**Comment by mikelesagitter** [comment]
![20240819_155125](https://github.com/user-attachments/assets/f103982a-3e4c-4dff-8301-ddc9be80a43a)
 It switches pin 11-3 and 14-16...

**Reply by AndyIreland** [reply]
yeah its for ford / forscan

**Reply by mikelesagitter** [reply]
I managed to do it!!!

I just soldered the right pins and all was good! So satisfying see the green lines on the right of ddt4all :D 
Now just the car does not start android auto but I'll solve this too! 

Thanks for your help ;)

**Comment by mikelesagitter** [comment]
PIC18F25K80 is not written on the chip
....
